crates/cli-tools/src/changelog.rs

```diff
@@ -266,19 +271,56 @@ struct Release {
     contents: BTreeMap<Severity, Vec<String>>,
 }
 
-impl Release {
-    fn push_content(&mut self, severity: Severity, content: &str) -> Result<()> {
-        self.contents.entry(severity).or_default().push(content.to_string());
-        Ok(())
-    }
-}
-
 impl From<Version> for Release {
     fn from(version: Version) -> Self {
         Release { version, contents: BTreeMap::new() }
     }
 }
 
+impl Release {
+    fn get_max_severity(&self) -> Severity {
+        for severity in [Severity::Major, Severity::Minor, Severity::Patch] {
+            if self.contents.contains_key(&severity) {
+                return severity;
+            }
+        }
+
+        Severity::Patch
+    }
+
+    fn increment_version(&mut self, severity: Severity) {
```

Comment:
I don't think this function will be useful once the test above is fixed. We probably want 2 functions:
- One that creates a pre-release from a release (just incrementing patch and adding `-git`).
- One that makes sure a pre-release is at least a given severity.

That last function could really just be:
- If version.major == 0 then reduce severity by one (major to minor, minor to patch, patch to patch).
- If severity is patch then do nothing and exit.
- If version.patch > 0 then increment version.minor and reset version.patch.
- If severity is minor then do nothing and exit.
- If version.minor > 0 then increment version.major and reset version.minor.
- Do nothing and exit (we know severity is major).

Reply:
I'm struggling to understand at a higher level. Could you explain how you see those 2 functions being used?

I think some examples of when we want to create a new release vs just increment version would be helpful.